Production/plugins/TauTupleProducer.cc

```diff
@@ -483,15 +498,95 @@ class TauTupleProducer : public edm::EDAnalyzer {
             tauTuple().ele_deltaEtaSuperClusterTrackAtVtx.push_back(ele->deltaEtaSuperClusterTrackAtVtx());
             tauTuple().ele_deltaEtaSeedClusterTrackAtCalo.push_back(ele->deltaEtaSeedClusterTrackAtCalo());
             tauTuple().ele_deltaEtaEleClusterTrackAtCalo.push_back(ele->deltaEtaEleClusterTrackAtCalo());
+            tauTuple().ele_deltaEtaSeedClusterTrackAtVtx.push_back(ele->deltaEtaSeedClusterTrackAtVtx());
             tauTuple().ele_deltaPhiEleClusterTrackAtCalo.push_back(ele->deltaPhiEleClusterTrackAtCalo());
             tauTuple().ele_deltaPhiSuperClusterTrackAtVtx.push_back(ele->deltaPhiSuperClusterTrackAtVtx());
             tauTuple().ele_deltaPhiSeedClusterTrackAtCalo.push_back(ele->deltaPhiSeedClusterTrackAtCalo());
 
-            tauTuple().ele_mvaInput_earlyBrem.push_back(ele->mvaInput().earlyBrem);
-            tauTuple().ele_mvaInput_lateBrem.push_back(ele->mvaInput().lateBrem);
-            tauTuple().ele_mvaInput_sigmaEtaEta.push_back(ele->mvaInput().sigmaEtaEta);
-            tauTuple().ele_mvaInput_hadEnergy.push_back(ele->mvaInput().hadEnergy);
-            tauTuple().ele_mvaInput_deltaEta.push_back(ele->mvaInput().deltaEta);
+            tauTuple().ele_mvaInput_earlyBrem.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->mvaInput().earlyBrem : default_int_value);
+            tauTuple().ele_mvaInput_lateBrem.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->mvaInput().lateBrem : default_int_value);
+            tauTuple().ele_mvaInput_sigmaEtaEta.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->mvaInput().sigmaEtaEta : default_value);
+            tauTuple().ele_mvaInput_hadEnergy.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->mvaInput().hadEnergy : default_value);
+            tauTuple().ele_mvaInput_deltaEta.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->mvaInput().deltaEta : default_value);
+
+            // shower shape variables are available for non-HGCal electrons with pt > 5
+            tauTuple().ele_sigmaIetaIphi.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->sigmaIetaIphi() : default_value);
```

Comment:
Similarly here, instead of repeating the pt cut value in each condition define `const bool hasShapeVars = !isHGCAL && ele->polarP4().pt() > 5;`

Reply:
Done